### PR TITLE
Gpl453 merged set control type

### PR DESCRIPTION
**Pull request by emrojo**
Closes #55 

Changes proposed in this pull request:

* Add control attributes to the sample if supplier sample id required them
* Changes in the payload sent to SS to create the tube rack
* Tubes will be an object instead of an array
* Purpose uuid and Study uuid needs to be provided from wrangler
* Size wont be needed as it will be selected with the purpose uuid

Depends on sanger/sequencescape#2683

### CODE REVIEW

File: wrangler/helpers/sample_helpers.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import re
+
+
+def position_contents_for(supplier_sample_id: str):
+    return {"content": sample_contents_for(supplier_sample_id)}
```
Comment (KatyTaylor) [inline wrangler/helpers/sample_helpers.py:5]: looks like `position_contents_for` isn't called anywhere?

File: tests/helpers/test_sample_helpers.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+def test_set_add_control_sample_if_present():
+    record = {"supplier_name": "is a positive control"}
+    add_control_sample_if_present(record)
+    assert record["control"] == True
+    assert record["control_type"] == "positive"
+
+
+def test_not_set_control_sample_if_not_present():
+    record = {"supplier_name": "is a very positive sample"}
```
Comment (pjvv) [inline tests/helpers/test_sample_helpers.py:40]: 😆

File: tests/helpers/test_sample_helpers.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    add_control_sample_if_present(record)
+    assert not "control" in record
```
Comment (pjvv) [inline tests/helpers/test_sample_helpers.py:42]: Do you mean to check `assert "control" not in record.keys()`? I might be misunderstanding your check.

File: wrangler/config/defaults.py [hunk 4]
```diff
@@ -3,7 +3,7 @@
 # MLWH details
 MLWH_DB_DBNAME = "mlwarehouse_dev"
 MLWH_DB_HOST = "localhost"
-MLWH_DB_PASSWORD = "root"
```
Comment (pjvv) [inline wrangler/config/defaults.py:6]: If you keep this as "root" here, you don't have to add anything to `development.py` or `test.py`.

File: wrangler/helpers/sample_helpers.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
+import re
+
+
+def sample_contents_for(supplier_sample_id: str):
+    return add_control_sample_if_present({"supplier_name": supplier_sample_id})
+
+
+def add_control_sample_if_present(sample_record: dict):
```
Comment (pjvv) [inline wrangler/helpers/sample_helpers.py:8]: To use dict as a type you need to import Dict from typing.

File: tests/helpers/test_sample_helpers.py (hunk 3 above)
Comment (emrojo) [reply tests/helpers/test_sample_helpers.py:42]: yes

File: wrangler/helpers/sample_helpers.py (hunk 1 above)
Comment (emrojo) [reply wrangler/helpers/sample_helpers.py]: Ok removed

File: tests/blueprints/test_labware.py [hunk 6]
```diff
@@ -3,11 +3,32 @@
...
+
 WRANGLE_URL = "/wrangle"
 VALID_BARCODE = "DN123"
 INVALID_BARCODE = "DN_invalid"
 LESS_TUBES_BARCODE = "DN_lesstubes"
 DIFF_TUBES_BARCODE = "DN_difftubes"
+SIZE48_BARCODE = "DN_48_valid"
+
+
+def mock_ss_path_with_uuid(app, mocked_responses, path, uuid):
```
[the remark this answers is not in the document]
Comment (emrojo) [reply tests/blueprints/test_labware.py]: Ok moved into fixture

File: tests/helpers/test_rack_helpers.py [hunk 7]
```diff
@@ -102,14 +104,38 @@ def test_parse_tube_rack_csv_ignores_no_read(app_db_less, client, tmpdir):
...
-    tubes = [
-        {"coordinate": "A01", "barcode": "TB123", "supplier_sample_id": "xyz123"},
-        {"coordinate": "A02", "barcode": "TB456", "supplier_sample_id": "xyz456"},
-    ]
+def test_create_tube_rack_body_with_size():
+    tubes = {
+        "A01": {"barcode": "TB123", "content": {"supplier_name": "xyz123"}},
+        "A02": {"barcode": "TB456", "content": {"supplier_name": "xyz456"}},
+    }
     size = 48
```
[the remark this answers is not in the document]
Comment (emrojo) [reply tests/helpers/test_rack_helpers.py:112]: Ok

File: wrangler/config/defaults.py (hunk 4 above)
Comment (emrojo) [reply wrangler/config/defaults.py]: Yes, changed back to root

File: wrangler/helpers/sample_helpers.py (hunk 5 above)
Comment (emrojo) [reply wrangler/helpers/sample_helpers.py]: Done

File: wrangler/config/defaults.py (hunk 4 above)
Comment (pjvv) [reply wrangler/config/defaults.py]: OK then you can remove the root from `development.py` and `test.py`.

**Pull request merged**